resources/assets/js/editor.js

```diff
@@ -148,7 +148,21 @@ function setupEditor(response) {
 }
 function addTab(file) {
     let span = document.createElement('span');
-    span.innerHTML = file.filename;
+    span.dataset.filename = file.filename;
+    let text = document.createTextNode(file.filename);
+    span.appendChild(text);
+    if(file.filename != 'main.py') {
+      let deleteButton = document.createElement('span');
+      deleteButton.innerHTML = '&times;';
+      deleteButton.onclick = function(this) {
+        let span = this.parentNode;
```

Comment:
span is already defined in this context so we don't need this line.

Reply:
Not very familiar with ES6 variables; thank you!